sips/sip-015/sip-015-network-upgrade.md

```diff
@@ -0,0 +1,1644 @@
+
+**Rationale:** This method facilitates authenticating principal types supplied
+by contract-calls or from decoded data.  Because principals must have the
+correct version byte in order to spend tokens, ensuring contracts can perform
+this check is of paramount importance to avoid loss-of-funds.
+
+**Examples:**
+
+```clarity
+(is-standard 'STB44HYPYAT2BB2QE513NSP81HTMYWBJP02HPGK6) ;; returns true on testnet and false on mainnet
+(is-standard 'STB44HYPYAT2BB2QE513NSP81HTMYWBJP02HPGK6.foo) ;; returns true on testnet and false on mainnet
+(is-standard 'SP3X6QWWETNBZWGBK6DRGTR1KX50S74D3433WDGJY) ;; returns true on mainnet and false on testnet
+(is-standard 'SP3X6QWWETNBZWGBK6DRGTR1KX50S74D3433WDGJY.foo) ;; returns true on mainnet and false on testnet
+(is-standard 'SZ2J6ZY48GV1EZ5V2V5RB9MP66SW86PYKKQ9H6DPR) ;; returns false on both mainnet and testnet
+```
+
+### New method: `to-consensus-buff`
+
+* **Input Signature:** `(to-consensus-buff x)`
+* **Output Signature:** `(optional buff))`
+
+The `to-consensus-buff` function is a special function that will serialize any
+Clarity value into a buffer, using the SIP-005 serialization of the
+Clarity value. Not all values can be serialized: some value's
+consensus serialization is too large to fit in a Clarity buffer (this
+is because of the type prefix in the consensus serialization).
+
+If the value cannot fit as serialized into the maximum buffer size,
+this returns `none`, otherwise, it will be
+`(some consensus-serialized-buffer)`. During type checking, the
+analyzed type of the result of this method will be the maximum possible
+consensus buffer length based on the inferred type of the supplied value.
+
+**Rationale:** This method is used to facilitate interactions with off-chain
+services, and to export data from a smart contract that can be consumed
+off-chain.
+
+**Examples:**
+
+```clarity
+(to-consensus-buff 1) ;; Returns (some 0x0000000000000000000000000000000001)
+(to-consensus-buff u1) ;; Returns (some 0x0100000000000000000000000000000001)
+(to-consensus-buff true) ;; Returns (some 0x03)
+(to-consensus-buff false) ;; Returns (some 0x04)
+(to-consensus-buff none) ;; Returns (some 0x09)
+(to-consensus-buff 'SZ2J6ZY48GV1EZ5V2V5RB9MP66SW86PYKKQ9H6DPR) ;; Returns (some 0x051fa46ff88886c2ef9762d970b4d2c63678835bd39d)
+(to-consensus-buff { abc: 3, def: 4 }) ;; Returns (some 0x0c00000002036162630000000000000000000000000000000003036465660000000000000000000000000000000004)
+```
+
+### New method: `from-consensus-buff`
+
+* **Input Signature:** `(from-consensus-buff type-signature buff)`
+* **Output Signature:** `(optional t)`
+
+The `from-consensus-buff` function is a special function that will deserialize a
+buffer into a Clarity value, using the SIP-005 serialization of the
+Clarity value. The type that `from-consensus-buff` tries to deserialize
+into is provided by `type-signature`. If it fails
+to deserialize the `buff` argument to this type, the method returns `none`.
```

Comment:
I'm sure input argument have hard size limit (1MB IIRC), and it should be specified here.

Reply:
The 1 MB limit applies to the `buff` type itself.  A clarity buffer cannot be greater than 1 MB in length as it is.  But I can mention that.

Reply:
Added